File: julia/atbash-cipher/atbash-cipher.jl
encode(input::AbstractString, chunk=5) = join(join.(Iterators.partition(filter(((!) ∘ isnothing), encode.(collect(lowercase(input)))), chunk)), " ")
encode(input::AbstractChar) = isnumeric(input) ? input : isletter(input) ? Char(219 - Int(input)) : nothing
decode(input) = encode(input, length(input))

File: csharp/atbash-cipher/AtbashCipher.cs
using System.Linq;

public static class AtbashCipher
{
    public static string Encode(string plainValue) => DoCipher(plainValue, 5);

    public static string Decode(string encodedValue) => DoCipher(encodedValue, encodedValue.Length);

    public static char DoCipher(char ch) => ch switch
    {
        _ when char.IsDigit(ch) => ch,
        _ when char.IsLetter(ch) => (char)(219 - char.ToLower(ch)),
        _ => default
    };

    public static string DoCipher(string str, int chunck) => string.Join(' ',
        str.Select(DoCipher)
           .Where(ch => ch != default)
           .Chunk(chunck)
           .Select(ch => string.Join("", ch)));
}
